### DISCUSSION TITLE
[windows] ssh-add -l Error connecting to agent: No such file or directory (ssh-agent)

### DISCUSSION THREAD

**Discussion by fcbry** [wezterm/wezterm#5541] [General]
I'm building from source, so I can't be certain when this issue manifested but it's there now. SSH agent stuff is broken. I lost a couple of hours trying to figure out what was wrong, but I've got a very simple workaround.  

I use ssh-agent, and windows, it's been working very well up until now, and I haven't been adding anything related to sockets into my `~/.ssh/config` file.

But, after deciding last friday afternoon to build wezterm from source, I found that ssh was broken, I narrowed it down to ssh agent issues, and the following canary: 

```
❯ ssh-add -l
Error connecting to agent: No such file or directory
```

The workaround:

```
❯ $ENV:SSH_AUTH_SOCK = $null
```

Now it works:
```
❯ ssh-add -l
3072 SHA256:K46sJXo20sbsQ6Cb/Upx3+sdh/Uq5srSxtHMy0OQK1U azuread\bryanhunt@bwin (RSA)
```

I'm going to add this to my Powershell startup file, but any hints on how to disable it on the wezterm side (or even how to create a wezterm build without the ssh features) would be appreciate. 

Thanks !

**Comment by fcbry** [comment]
Ah, I can disable it with :

```
config.mux_enable_ssh_agent = false
```

Works great now. Feel free to close this thread. Might be a good idea to set it this way on windows 11 though.

**Reply by DeadlySquad13** [reply]
Sorry for the reply to the old discussion but thank you very much for solution! Have been struggling with it after updating wezterm recently and didn't know the reason.
